crates/sui-graphql-rpc/src/context_data/pg_backend.rs

```diff
@@ -631,16 +632,38 @@ impl PgQueryExecutor for PgManager {
         self.inner
             .spawn_blocking(move |this| {
                 let query = query_builder_fn()?;
-                let explain_result: String = this
-                    .run_query(|conn| query.explain().get_result(conn))
-                    .map_err(|e| Error::Internal(e.to_string()))?;
-                let cost = extract_cost(&explain_result)?;
-                if cost > max_db_query_cost as f64 {
-                    return Err(DbValidationError::QueryCostExceeded(
-                        cost as u64,
-                        max_db_query_cost,
-                    )
-                    .into());
+                let explain_result: Option<String> =
+                    match this.run_query(|conn| query.explain().get_result(conn)) {
+                        Ok(result) => Some(result),
+                        Err(e) => {
+                            info!(
+                                target: "async-graphql",
+                                "Failed to get explain result: {}", e
+                            );
+                            None
+                        }
+                    };
+
+                if let Some(explain_result) = explain_result {
+                    let cost = match extract_cost(&explain_result) {
+                        Ok(cost) => Some(cost),
+                        Err(e) => {
+                            info!(
+                                target: "async-graphql",
+                                "Failed to get cost from explain result: {}", e
+                            );
+                            None
+                        }
+                    };
+
+                    if let Some(cost) = cost {
+                        if cost > max_db_query_cost as f64 {
+                            info!(
+                                target: "async-graphql",
+                                cost,
+                            );
+                        }
+                    }
                 }
```

Comment:
- You may find the [tap](https://docs.rs/tap/1.0.1/tap/trait.TapFallible.html#tymethod.tap_err ) crate useful (we already use it in our codebase).
- Use different log levels other than `info` so you can modulate how much signal you get from these logs on the fly.
- (I folded in the suggestion to use a different target).

```suggestion
                let explain_result = this
                    .run_query(|conn| query.explain().get_result(conn))
                    .tap_err(|e| warn!(
                        target: "explain",
                        "Failed to get explain result: {}", e
                    ))?;

                let cost = extract_cost(&explain_result)
                    .tap_err(|e| warn!(
                        target: "explain",
                        "Failed to get cost from explain result: {}", e
                    ))?;
                    
                if cost > max_db_query_cost as f64 {
                    warn!(target: "explain", "Explain cost exceeded: {cost}");
                } else {
                    info!(target: "explain", "Explain cost: {cost}");
                }
```